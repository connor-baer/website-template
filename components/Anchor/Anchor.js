import React from 'react';
import styled from '@emotion/styled';
import { css } from '@emotion/core';

import Link from '../Link';

const baseStyles = ({ theme }) => css`
  color: ${theme.colors.p500};
  transition: color 0.1s ease-in-out, border 0.1s ease-in-out,
    background-color 0.1s ease-in-out, outline 0.1s ease-in-out;

  &:hover {
    color: ${theme.colors.p900};
  }

  &:focus {
    outline: thin dotted currentColor;
    outline-offset: 0.25em;
  }

  &:active {
    color: ${theme.colors.p500};
  }
`;

const A = styled('a')(baseStyles);

export default function Anchor({ children, className, ...rest }) {
  return (
    <Link {...rest} passHref>
      <A className={className}>{children}</A>
    </Link>
  );
}
