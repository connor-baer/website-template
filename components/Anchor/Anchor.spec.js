import React from 'react';

import Anchor from './Anchor';

describe('Anchor', () => {
  function renderAnchor(fn, props = {}) {
    return fn(<Anchor {...props} />);
  }

  it('should render with default styles', () => {
    const href = 'https://connorbaer.co';
    const children = 'label';
    const { container } = renderAnchor(render, { href, children });
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should meet basic accessibility guidelines', async () => {
    const href = 'https://connorbaer.co';
    const children = 'label';
    const html = renderAnchor(renderToHtml, { href, children });
    const result = await axe(html);
    expect(result).toHaveNoViolations();
  });
});
