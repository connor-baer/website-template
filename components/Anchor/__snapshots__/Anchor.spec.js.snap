// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor should render with default styles 1`] = `
.circuit-0 {
  color: #f06595;
  -webkit-transition: color 0.1s ease-in-out,border 0.1s ease-in-out,background-color 0.1s ease-in-out,outline 0.1s ease-in-out;
  transition: color 0.1s ease-in-out,border 0.1s ease-in-out,background-color 0.1s ease-in-out,outline 0.1s ease-in-out;
}

.circuit-0:hover {
  color: #a61e4d;
}

.circuit-0:focus {
  outline: thin dotted currentColor;
  outline-offset: 0.25em;
}

.circuit-0:active {
  color: #f06595;
}

<a
  class="circuit-0 circuit-1"
  href="https://connorbaer.co/"
>
  label
</a>
`;
